src/components/controls/MediaStreamControl.tsx

```diff
@@ -48,58 +50,54 @@ export class MediaStreamControl extends React.PureComponent<IMediaControlProps,
         )
     }
 
-    private getAudioEnabled(): boolean {
-        const track = this.props.stream.getAudioTracks()[0];
-        return track ? track.enabled : false;
-    }
-
-    private setAudioEnabled(audioEnabled: boolean): void {
+    private toggleAudioEnabled(): void {
         if (!this.verifyStream()) {
             return;
         }
-        const track = this.props.stream.getAudioTracks()[0];
+
+        const { audioEnabled } = this.props.stream;
+        this.props.onAudioEnabledChange(!audioEnabled);
```

Comment:
We should add a TODO here for a better design. This allows two sources of truth for audio/video enabled statuses.